### PR TITLE
Documentation: add table of contents to the api guide

### PR DESCRIPTION
**Pull request by kelseyhightower**

### CODE REVIEW

File: Documentation/api.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
 # etcd API
 
+- [Key Space Operations](#key-space-operations)
```
Comment (xiang90) [inline Documentation/api.md]: i think the table is too huge.

how about combine some of them? for example, list dir/delete dir -> dir opeartion

Comment (kelseyhightower) [inline Documentation/api.md]: I've combined the directory operations, but I think we need to spit this doc up. Consider this the first step in improving readability.

Comment (xiang90) [inline Documentation/api.md]: i still do not quite like the big tables actually. but it might be only me.

/cc @yichengq @barakmich

Comment (yichengq) [inline Documentation/api.md]: I also think it contains too many items and hard to follow.
A good categorized API so far is here: https://github.com/coreos/etcd/blob/master/client/keys.go#L93
It lists the most common operations at the first level, and give more options in the secondary level.

**Comment by kelseyhightower** [comment]
@yichengq @xiang90 Here is what I consider great API docs: https://developer.github.com/v3/

What I like about them is that I can quickly find all the things the API can do, even for some one new to the GitHub API everything is easy to find.

**Comment by kelseyhightower** [comment]
@yichengq @xiang90 I also like the way godocs adds a TOC at the top of every package: http://golang.org/pkg/io/ioutil/

Notice how everything function the package supports is easy to discover because of the TOC at the top of the page. As far as length goes, we should consider that these docs serve as reference for the user, both new and experienced.

Most of our users do not remember the etcd API, so a detailed reference (TOC) makes all the difference.

**Comment by kelseyhightower** [comment]
@robszumski ^

**Comment by xiang90** [comment]
@kelseyhightower I like TOC on top. I hope we can have a one level TOC with limited number( < 8? ) of items.

**Comment by robszumski** [comment]
This will conflict with the sidebar that we have on coreos.com, but a lot of people that consume the docs do so on GitHub.

**Comment by kelseyhightower** [comment]
I'm going to close this since it seems we need to have a larger discussion on docs for etcd.
